docs/getting-started.md

```diff
@@ -0,0 +1,325 @@
+c.JupyterHub.ssl_cert = '/path/to/my.cert'
+```
+
+There are two other aspects of JupyterHub network security.
+The Hub authenticates its requests to the proxy via an environment variable,
+`CONFIGPROXY_AUTH_TOKEN`. If you want to be able to start or restart the proxy
+or Hub independently of each other (not always necessary),
+you must set this environment variable before starting the server:
+
+```bash
+export CONFIGPROXY_AUTH_TOKEN=`openssl rand -hex 32`
+```
+
+If you don't set this, the Hub will generate a random key itself,
+which means that any time you restart the Hub you **must also restart the proxy**.
+If the proxy is a subprocess of the Hub, this should happen automatically.
+
+The cookie secret is another key, used to encrypt the cookies used for authentication.
+If this value changes for the Hub, all single-user servers must also be restarted.
+Normally, this value is stored in the file `jupyterhub_cookie_secret`, which can be specified with:
+
+```python
+c.JupyterHub.cookie_secret_file = '/path/to/cookie_secret'
+```
+
+If the cookie secret file doesn't exist when the Hub starts,
+a new cookie secret is generated and stored in the file.
+
+If you would like to avoid the need for files,
+the value can be loaded from the `JPY_COOKIE_SECRET` env variable:
+
+```bash
+export JPY_COOKIE_SECRET=`openssl rand -hex 1024`
+```
+
+
+## Configuring Authentication
+
+The default Authenticator uses [PAM][] to authenticate system users with their username and password.
+The default behavior of this Authenticator is to allow any users with a password on the system to login.
+You can restrict which users are allowed to login with `Authenticator.whitelist`:
+
+```python
+c.Authenticator.whitelist = {'mal', 'zoe', 'inara', 'kaylee'}
+```
+
+After starting the server, you can add and remove users in the whitelist via the `admin` panel,
+which brings us to...
+
+```python
+c.JupyterHub.admin_users = {'mal', 'zoe'}
+```
+
+Any users in the admin list are automatically added to the whitelist, if they are not already present.
+
+Admin users have the ability to take actions on users' behalf,
+such as stopping and restarting their servers, and adding and removing new users.
+If `JupyterHub.admin_access` is True (not default),
+then admin users have permission to log in *as other users* on their respective machines,
+for debugging. **You should make sure your users know if admin_access is enabled.**
+
+### adding and removing users
+
+The default PAMAuthenticator is one case of a special kind of authenticator,
+called a LocalAuthenticator,
+indicating that it manages users on the local system.
+When you add a user to the Hub, a LocalAuthenticator checks if that user already exists.
+Normally, there will be an error telling you that the user doesn't exist.
+If you set the config value
+
+```python
+c.LocalAuthenticator.create_system_users = True
+```
+
+however, adding a user to the Hub that doesn't already exist on the system will result
```

Comment:
added at the top of this section, since it is generic to adding users no matter what the Authenticator.